### PR TITLE
Wiring again

### PR DESCRIPTION
**Pull request by chrisforbes**
Nowhere near complete, but this is the start of the new system.

### CODE REVIEW

**Comment by chrisforbes** [comment]
No, there's no need to implement unselect unless your tool actually dirties the world prior to use().

File: src/block.h [hunk 1]
```diff
@@ -47,6 +47,7 @@ struct block {
     block_type type = block_untouched;
 
     surface_type surfs[face_count]{};
+    bool has_wire[face_count]{};
```
Comment (RobotCaleb) [inline src/block.h]: How well does this extend to not just one wire? What are our options: we either break the world up into smaller chunks for wiring or we mark how many(?) wires are on a surface.

File: src/tools/wiring.cc [hunk 2]
```diff
@@ -15,17 +15,127 @@ extern ship_space *ship;
...
+        return pos == other.pos && face == other.face;
+    }
+
+    bool operator!=(wire_pos const & other) const {
+        return pos != other.pos || face != other.face;
+    }
+
+    struct hash {
+        size_t operator()(wire_pos const &w) const {
+            return ivec3_hash()(w.pos) ^ std::hash<unsigned>()(w.face);
```
Comment (RobotCaleb) [inline src/tools/wiring.cc]: Will this not collide?

**Comment by chrisforbes** [comment]
Adjusted penalty, added missing vox source to first commit

**Comment by chrisforbes** [comment]
Pushing based on @RobotCaleb's +1 on IRC.

**Pull request merged**